src/main/kotlin/org/rust/lang/core/completion/LookupElements.kt

```diff
@@ -291,18 +302,30 @@ open class RsDefaultInsertHandler : InsertHandler<LookupElement> {
             is RsStructItem -> appendSemicolon(context, curUseItem)
 
             is RsFunction -> {
-                if (curUseItem != null) {
-                    appendSemicolon(context, curUseItem)
-                } else {
-                    val isMethodCall = context.getElementOfType<RsMethodOrField>() != null
-                    if (!context.alreadyHasCallParens && !context.isInMetaItem) {
-                        document.insertString(context.selectionEndOffset, "()")
-                        context.doNotAddOpenParenCompletionChar()
+                when {
```

Comment:
Btw it is possible to avoid one indent by moving `when` on the previous line and removing braces, i.e. `is RsFunction -> when { ...`